shared/src/androidMain/kotlin/org/jetbrains/kotlinconf/AndroidNotificationService.kt

```diff
@@ -9,43 +9,42 @@ import android.content.Context
 import android.content.Intent
 import android.content.pm.PackageManager
 import android.os.Build
-import androidx.activity.result.ActivityResultLauncher
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationManagerCompat
 import androidx.core.content.ContextCompat
 import androidx.core.content.getSystemService
-import kotlinx.coroutines.channels.Channel
 import kotlinx.datetime.LocalDateTime
 import kotlinx.datetime.toInstant
 import org.jetbrains.kotlinconf.utils.Logger
-import org.koin.mp.KoinPlatformTools
+import org.koin.mp.KoinPlatform
 
-private const val LOG_TAG = "AndroidNotificationService"
+const val EXTRA_NOTIFICATION_ID = "notificationId"
 private const val EXTRA_TITLE = "title"
 private const val EXTRA_MESSAGE = "message"
-const val EXTRA_NOTIFICATION_ID = "notificationId"
-private const val EXTRA_ICON_ID = "iconId"
 private const val NOTIFICATION_CHANNEL_ID = "channel_all_notifications"
 private const val ACTION_SHOW_NOTIFICATION = "org.jetbrains.kotlinconf.SHOW_NOTIFICATION"
 
 class AndroidNotificationService(
     private val timeProvider: TimeProvider,
     private val context: Context,
     private val iconId: Int,
-    private val permissionResult: Channel<Boolean>,
-    private val permissionLauncher: ActivityResultLauncher<Array<String>>,
     private val logger: Logger,
 ) : NotificationService {
 
+    companion object {
+        private const val LOG_TAG = "AndroidNotificationService"
+    }
+
     private val notificationManager = NotificationManagerCompat.from(context)
     private val alarmManager = context.getSystemService<AlarmManager>()
 
     override suspend fun requestPermission(): Boolean {
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU) return true
 
         val permissions = arrayOf(Manifest.permission.POST_NOTIFICATIONS, Manifest.permission.USE_EXACT_ALARM)
-        permissionLauncher.launch(permissions)
-        return permissionResult.receive()
+
+        val permissionHandler = KoinPlatform.getKoin().getOrNull<PermissionHandler>()
```

Comment:
You can use the `getKoin()` Service extension instead if you want.

Reply:
Despite the name, this is not an `android.app.service`, so I don't think we can do that here. Does `getKoin()` perform the exact same `KoinPlatform.getKoin()` internally anyway?